Automattic/simplenote-ios
# Login with Passkeys!!

### Fix
To improve the security of Simplenote accounts we are trying to move away from email/password authentication as it is insecure at best.  One of the things we are implementing is Passkeys.  This will allow users to create and store passkeys on their apple device and when they go to log into their account rather than using an email and password they only need to put in their email and then use the local biometrics to receive a passkey and then authenticate into Simplenote.

This has a bunch of security benefits and should also help users keep from losing their passwords.

Sign up for passkeys:


https://github.com/Automattic/simplenote-ios/assets/22015538/18a4b140-e892-4691-86ad-9d31c6d89334



Login with passkeys:


https://github.com/Automattic/simplenote-ios/assets/22015538/4797e3bc-4ea6-4ea8-83f6-77e9ca9ea2f0



### Test
1. Clean install Simplenote on device (can't access biometrics on sim)  Tap on login and choose to login with email and password
2. Once logged in go to the menu -> Settings and in the account section you will see a "Add Passkey Authentication" button.  Press that.  After a few seconds you will receive a notice from device asking if you want to create a passkey.  Confirm and wait a few seconds.  (NOTE: Currently there is no spinner or success/failure indication, that will be in a future PR)
3. Log out of Simplenote
4. Tap on the Login button and choose Login with Passkeys
5. Enter your email and tap on the Login with Passkeys Button
6. after a few seconds you will be prompted by device if you want to login with passkeys, confirm

- [x] Confirm you are able to log into your Simplenote account.

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer is required to review these changes, but anyone can perform the review.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
> `RELEASE-NOTES.txt` was updated in 0c4129 with:
> 
> > Added Passkey authentication support
